lib/options.go

```diff
@@ -164,6 +175,29 @@ func (c *TLSAuth) Certificate() (*tls.Certificate, error) {
 	return c.certificate, nil
 }
 
+func parsePrivateKey(privKey, password string) ([]byte, error) {
+	key := []byte(privKey)
+	block, _ := pem.Decode(key)
+	if block == nil {
+		return nil, fmt.Errorf("failed to decode PEM key")
+	}
+	blockType := block.Type
+	if blockType == "ENCRYPTED PRIVATE KEY" {
+		return nil, fmt.Errorf("encrypted pkcs8 formatted key is not supported")
+	}
+	if encrypted := x509.IsEncryptedPEMBlock(block); encrypted {
+		decryptedKey, err := x509.DecryptPEMBlock(block, []byte(password))
+		if err != nil {
+			return nil, err
+		}
+		key = pem.EncodeToMemory(&pem.Block{
+			Type:  blockType,
+			Bytes: decryptedKey,
+		})
+	}
```

Comment:
Can we at least get a warning that "password" is provided byt the private key isn't encrypted?

Reply:
see https://github.com/grafana/k6/pull/2488/files#r849421221 we can't do it